master/internal/rm/agentrm/agent_state_test.go

```diff
@@ -322,6 +323,62 @@ func Test_agentState_checkAgentStartedDevicesMatch(t *testing.T) {
 	}
 }
 
+func Test_agentState_checkAgentResourcePoolMatch(t *testing.T) {
+	tests := []struct {
+		name            string
+		state           agentState
+		agentStarted    *aproto.AgentStarted
+		wantErrContains string
+	}{
+		{
+			name: "resource pool name match",
+			state: agentState{
+				resourcePoolName: "pool1",
+			},
+			agentStarted: &aproto.AgentStarted{
+				ResourcePoolName: "pool1",
+			},
+			wantErrContains: "",
+		},
+		{
+			name: "resource pool name is missing",
+			state: agentState{
+				resourcePoolName: "pool1",
+			},
+			agentStarted:    &aproto.AgentStarted{ResourcePoolName: defaultResourcePoolName},
+			wantErrContains: "resource pool has changed",
+		},
+		{
+			name: "resource pool name is missing",
+			state: agentState{
+				resourcePoolName: defaultResourcePoolName,
+			},
+			agentStarted:    &aproto.AgentStarted{ResourcePoolName: "pool1"},
+			wantErrContains: "resource pool has changed",
+		},
+		{
+			name: "mismatched resource pool name",
+			state: agentState{
+				resourcePoolName: "pool1",
+			},
+			agentStarted: &aproto.AgentStarted{
+				ResourcePoolName: "pool2",
+			},
+			wantErrContains: "resource pool has changed",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := tt.state.checkAgentResourcePoolMatch(tt.agentStarted)
+			if tt.wantErrContains == "" {
```

Comment:
I love this use of `""` as a way to do this without a pointer since we'll never want to check whether an empty string is a substring of an error message.